frontend/src/Components/PrivateRoute.js

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { Route, Redirect, withRouter } from "react-router-dom";
+import { connect } from 'react-redux'
+
+const PrivateRoute = ({ component: Component, ...rest }) => {
+  return (
+    <Route
+      {...rest}
+      render={props =>
+        localStorage.getItem("jwt") ? (
```

Comment:
Not as big of a deal here, but with larger projects, you'll actually want to confirm the JWT belongs to who you think it belongs to before grabbing any local data for them.